### PR TITLE
ui: use `Button` component from `HDS`

### PR DESCRIPTION
**Pull request by ChaiWithJai**
This PR updates the `jobs.run` view to use the `Button` style and component from HDS. All previous behavior has been preserved:

## Previous Behavior
- On cancel, the form resets
- On incorrect submission, a red message appears (no reset option)
- Invalid output like `1` will not trigger an invalidation error
- On the first time hitting "plan" you'll get an info message (based on `localStorage`). This message will no longer appear after the first time hitting "plan"
- On hitting "run" you will navigate to the new job's detail page (no error handling on this -- see point #3)

## Initial View
Screen Shot 2022-12-21 at 4 30 08 PM

## On first time using (based on `localStorage`)
Screen Shot 2022-12-21 at 4 30 59 PM

### CODE REVIEW

File: ui/app/components/job-editor.js [hunk 1]
```diff
@@ -105,6 +105,7 @@ export default class JobEditor extends Component {
   })
   submit;
 
+  @action
```
Comment (ChaiWithJai) [inline ui/app/components/job-editor.js:108]: `explainer`: Needed for `this` reference on invocation from template.

File: ui/app/templates/components/job-editor.hbs [hunk 2]
```diff
@@ -22,36 +22,11 @@
...
     <div class="boxed-section-head">
       Job Definition
-      {{#if this.cancelable}}
-        <button
-          class="button is-light is-compact pull-right"
-          onclick={{action this.onCancel}}
-          data-test-cancel-editing
-          type="button"
-        >Cancel</button>
-      {{/if}}
```
Comment (ChaiWithJai) [inline ui/app/templates/components/job-editor.hbs:54]: This was dead code.

File: ui/app/templates/components/job-editor.hbs [hunk 3]
```diff
@@ -110,7 +88,7 @@
   </div>
   <div
     class="boxed-section
-      {{if this.planOutput.failedTGAllocs 'is-warning' 'is-primary'}}"
+      {{if this.planOutput.failedTGAllocs 'is-warning'}}"
```
Comment (ChaiWithJai) [inline ui/app/templates/components/job-editor.hbs:91]: Breaks `a11y` color rule.

**Review by philrenaud — APPROVED** [review]
Approving to move things along; buttons look nice! Noticed a couple things though:

[Object object] error: 
image
^--- in mirage when just passing a nonsense string to plan

File: ui/app/templates/components/job-editor.hbs [hunk 4]
```diff
@@ -160,19 +146,8 @@
...
-    <button
-      class="button is-light"
-      type="button"
-      onclick={{action this.reset}}
-      data-test-cancel
-    >Cancel</button>
-  </div>
+  <Hds::ButtonSet class="is-associative">
+    <Hds::Button @size="small" @text="Run" {{on "click" (perform this.submit)}} disabled={{this.submit.isRunning}} data-test-run />
+    <Hds::Button @size="small" @color="critical" @text="Cancel" {{on "click" this.reset}} disabled={{this.submit.isRunning}} data-test-cancel />
```
[the remark this answers is not in the document]
Comment (ChaiWithJai) [reply ui/app/templates/components/job-editor.hbs:151]: So you think we wanna go with medium?

[the remark this answers is not in the document]
Comment (philrenaud) [reply ui/app/templates/components/job-editor.hbs:151]: Yeah, let's try to match font size with the rest of the app if we can.

File: ui/app/templates/components/job-editor.hbs (hunk 3 above)
Comment (philrenaud) [reply ui/app/templates/components/job-editor.hbs:99]: Okay; let's work with design to find something that indicates a success. We'll have this situation come up in the future in other contexts as well, I suspect, and worth setting a good precedent here.

**Pull request merged**